engine/execution/computation/computer/uploader/uploader_test.go

```diff
@@ -95,6 +95,8 @@ func Test_AsyncUploader(t *testing.T) {
 		require.Equal(t, 3, callCount)
 	})
 
+	time.Sleep(1 * time.Second)
+
```

Comment:
This seems quite a bit unrelated? Have you seen this being flaky? /cc @m4ksio

Reply:
> Concept ACK.
> 
> Idle musings:
> How often do we call this in contexts requiring deep equality, I wonder? Is it possible that we ferry around a header by reference far enough that we forget how to get its ID, rather than actually compare blocks from different source? In other terms, could we try pointer equality optimistically, and degrade to the full comparison when the pointers don't match?

Good idea.  I'll put that in the next version